docs/configuration.md

```diff
@@ -0,0 +1,57 @@
+# Configuration
+
+## Configuration methods
+
+Use the environment variables in the tables below to configure the [automatic instrumentation binary](https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/main/cli/main.go).
+
+Alternatively, you can add support for additional or different configurations by building your own Go automatic instrumentation binary using the [OpenTelemetry Go Automatic Instrumentation package](https://pkg.go.dev/go.opentelemetry.io/auto).
+
+## Global settings
```

Comment:
Maybe worth adding a section about instrumentation specific configuration? Currently the only example we have that I can think of is to include the SQL query in database/sql. But we are planning to have more in the future. If it is out of scope for this PR it's fine.

Reply:
Can you share what config option is missing so that it can be added (here or separate PR)?

Reply:
`OTEL_GO_AUTO_INCLUDE_DB_STATEMENT`

Reply:
I think it might be clearer to have a separate section for instrumentation configuration (as the above option is not configuring the exporter but the instrumentation itself - collecting an extra attribute). Currently, the above env var is the only config option we have under that category so we can change it in the future once we'll have more options.

Reply:
IMO, it would be fine to create the section now with just the one entry. It might look a little funny to have just one config option, but I'd rather that than bury the env var in the wrong place and potentially confuse people about what it's for. But I don't have a ton of context, so we can go with whatever you recommend.

Reply:
You can use https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/config.md#instrumentation-options for inspiration.

Reply:
I moved the env var to a new Instrumentation options section. As we fill in the new section with more env vars, we can add similar structure and detail to what @pellared linked to in the dotnet docs.